Demo/src/main/java/com/paypal/android/viewmodels/PayPalNativeViewModel.kt

```diff
@@ -0,0 +1,170 @@
+package com.paypal.android.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.paypal.android.usecase.GetAccessTokenUseCase
+import com.paypal.android.usecase.GetApprovalSessionIdActionUseCase
+import com.paypal.android.usecase.GetBillingAgreementTokenUseCase
+import com.paypal.android.usecase.GetOrderIdUseCase
+import com.paypal.android.usecase.GetOrderUseCase
+import com.paypal.android.PayPalConfigConstants
+import com.paypal.android.checkout.PayPalCheckoutError
+import com.paypal.android.checkout.PayPalCheckoutListener
+import com.paypal.android.checkout.PayPalCheckoutResult
+import com.paypal.android.checkout.PayPalClient
+import com.paypal.android.core.CoreConfig
+import com.paypal.android.core.PayPalSDKError
+import com.paypal.checkout.createorder.CreateOrder
+import com.paypal.checkout.error.ErrorInfo
+import com.paypal.checkout.shipping.ShippingChangeActions
+import com.paypal.checkout.shipping.ShippingChangeData
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import retrofit2.HttpException
+import javax.inject.Inject
+
+@HiltViewModel
+class PayPalNativeViewModel @Inject constructor(
+    private val getBillingAgreementTokenUseCase: GetBillingAgreementTokenUseCase,
+    private val getAccessTokenUseCase: GetAccessTokenUseCase,
+    private val getOrderIdUseCase: GetOrderIdUseCase,
+    private val getApprovalSessionIdActionUseCase: GetApprovalSessionIdActionUseCase,
+    application: Application
+) : AndroidViewModel(application) {
+
+    private val getOrderUseCase: GetOrderUseCase = GetOrderUseCase()
```

Comment:
this can be injected as constructor param

Reply:
@kgangineni i had to remove them all from the constructor because `detekt` was failing with `LongParameterList`, so I inject them now as variables

Reply:
in these type of cases, it is better to suppress detekt because this is genuine case of we requiring more paramters I see u already made changes to all dependencies to be `lateinit var` that's good too